tools/azure-pipelines/cache_docker_images.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+#
+# To reduce the number of pull requests we make against dockerhub, this script
+# saves the docker images we typically pull during a testing cycle. This typically
+# includes testcontainer images, kafka, elasticearch, etc.
+#
+echo "=============================================================================="
+echo "Caching Testing Docker Images"
+echo "=============================================================================="
+echo ""
+
+if [ -z "$TESTCONTAINER_CACHE_FOLDER" ]
+then
+      echo "\$TESTCONTAINER_CACHE_FOLDER must be set to cache the testing docker images. Exiting"
+      exit 1
+fi
+
+mkdir -p "${TESTCONTAINER_CACHE_FOLDER}"
+
+# This is the pattern that determines which containers we save.
+DOCKER_IMAGE_CACHE_PATTERN="testcontainers|kafka|elasticsearch|postgres|mysql"
```

Comment:
Nothing we need to fix now, but long-term it could be neat to derive this from `TestContainerVersions.java`.